src/dft/backends/rocfft/commit.cpp

```diff
@@ -259,12 +260,30 @@ class rocfft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::reverse(stride_vecs.vec_b.begin(), stride_vecs.vec_b.end());
         stride_vecs.vec_b.pop_back(); // Offset is not included.
 
-        rocfft_plan_description plan_desc;
-        if (rocfft_plan_description_create(&plan_desc) != rocfft_status_success) {
+        // This workaround is needed due to a confirmed issue in rocFFT from version
+        // 1.0.23 to 1.0.30. Those rocFFT version correspond to rocm version from
+        // 5.6.0 to 6.3.0.
+        // Link to rocFFT issue: https://github.com/ROCm/rocFFT/issues/507
+        if constexpr (rocfft_version_major == 1 && rocfft_version_minor == 0 &&
+                      (rocfft_version_patch > 22 && rocfft_version_patch < 31)) {
+            if (dom == dft::domain::COMPLEX &&
+                config_values.placement == dft::config_value::NOT_INPLACE && dimensions > 2) {
```

Comment:
Are we certain this is limited to `dimensions > 2`? Is a 2D case like
`lengths = {A, B}`
`inStrides = {B 1}`
`outStrides = {1, A}`
present in the test base?
Possibly relevant as well are 1D batch cases like
`length = N;`
`batch = M;`
`inStrides = 1; inDistance = N;`
`outStrides = M; outDistance = 1;`

Reply:
Thank you for your review @raphael-egan 
The 2D configuration above is not directly tested. If added, rocfft backend skips the tests with the following output:
```bash
Skipping test because: "oneMKL: DFT/commit: function is not implemented rocfft requires a stride to be divisible by all dimensions associated with smaller strides!"
```
If a configuration that respect inputs requirement is provided, tests are passed. Similar configurations are already in the test suite, so I don't think adding this test is a priority.

For 1D there isn't a test that reproduces what you suggested. I added it to check. It is computed correctly with the input above, although the `outDistance` is computed internally to perform the operation correctly. In any case, both configuration do not cause any issue with rocFFT.

Therefore, I wouldn't change the if-statement condition

Reply:
Given that the exception is thrown by this project and not by rocFFT itself, I would recommend to add a comment like
`// rocFFT's functional status for problems like cfoA:B:1xB:1:A is unknown as of [commit hash] (due to project's implementation preventing testing thereof)`

Unless an explicit mention of that as a known limitation is found in rocFFT's documentation/release notes (I could not find any), it looks to me that the check triggering this exception to be thrown is present here to ensure consistency of subsequent stride-related implementation steps (sorting operations, in particular).

Reply:
I agree, added comment in [e84a5b2](https://github.com/oneapi-src/oneMKL/pull/563/commits/e84a5b28c44e6823e5a08d1a06e40b3171858666)